### PR TITLE
New features, bug fixes, code improvements

### PR DESCRIPTION
**Pull request by frinkelpi**
- [added] Display train numbers
- [added] Date parsing (#10) and support of time without columns (#17)
- [changed] Improve display by using texttable
- [changed] Improve code modularity, use argparse for argument parsing
- [fixed] Python 2 and 3 bugfixes. Tests using nose2. Removed dependency on envoy.

### CODE REVIEW

**Comment by dbrgn** [comment]
Thanks! That's quite some changes :) I'll review them this week, hopefully.

**Comment by dbrgn** [comment]
But first of all, could you include texttable via PyPI instead of using submodules?

**Comment by frinkelpi** [comment]
Sure, that's because I had to submit a fix, but the module has now been updated on pypi, and I'll drop the submodule dependency.

**Comment by dbrgn** [comment]
I'm getting an exception on the following query:

```
$ python -m fahrplan.main von burgdorf nach rüti via bern
Traceback (most recent call last):
  File "/usr/lib64/python3.5/runpy.py", line 184, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib64/python3.5/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/danilo/Projects/fahrplan/fahrplan/main.py", line 109, in <module>
    main()
  File "/home/danilo/Projects/fahrplan/fahrplan/main.py", line 95, in main
    data = getConnections(args, (output_format == Formats.FULL), proxy_host)
  File "/home/danilo/Projects/fahrplan/fahrplan/API.py", line 134, in getConnections
    data["connections"] = [_parse_connection(c, include_sections) for c in data["connections"]]
  File "/home/danilo/Projects/fahrplan/fahrplan/API.py", line 134, in <listcomp>
    data["connections"] = [_parse_connection(c, include_sections) for c in data["connections"]]
  File "/home/danilo/Projects/fahrplan/fahrplan/API.py", line 115, in _parse_connection
    section = _parse_section(con_sections[0], connection)
  File "/home/danilo/Projects/fahrplan/fahrplan/API.py", line 62, in _parse_section
    section['travelwith'] = journey["name"]
TypeError: 'NoneType' object is not subscriptable
```

**Comment by dbrgn** [comment]
And sorry for not reviewing sooner. In the future, sending smaller pull requests makes it easier and faster to review and merge :)

File: fahrplan/API.py [hunk 1]
```diff
@@ -59,10 +59,10 @@ def _parse_section(con_section, connection):
     section = {}
     section['station_from'] = departure['station']['name']
     section['station_to'] = arrival['station']['name']
-    section['travelwith'] = journey["name"]
+    section['travelwith'] = journey["name"] if journey!=None else ""
```
Comment (dbrgn) [inline fahrplan/API.py]: As `None` is a singleton in Python, the comparison must be `if journey is not None`.

**Comment by dbrgn** [comment]
Thanks! If you have further improvements, small pull requests help :)

**Pull request closed without merging**